dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcRequestHandlerMapping.java

```diff
@@ -42,9 +43,16 @@ protected boolean supportContentType(String contentType) {
 
     @Override
     protected void determineHttpMessageCodec(RequestHandler handler, URL url, HttpRequest request) {
-        HttpMessageCodec codec = CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
-        handler.setHttpMessageDecoder(codec);
-        handler.setHttpMessageEncoder(codec);
+        GrpcCompositeCodec grpcCompositeCodec =
+                (GrpcCompositeCodec) CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
+        MethodDescriptor methodDescriptor = DescriptorUtils.findMethodDescriptor(
+                handler.getServiceDescriptor(), handler.getMethodName(), handler.isHasStub());
+        if (methodDescriptor != null) {
+            handler.setMethodDescriptor(methodDescriptor);
+            grpcCompositeCodec.loadPackableMethod(methodDescriptor);
```

Comment:
**Security Issue**: The code dynamically loads a MethodDescriptor based on request data without proper validation, potentially allowing for unauthorized method execution or access control bypass. <br> **Fix**: Implement strict validation of the request data used to find the MethodDescriptor. Ensure that the method access is authorized for the requesting user before setting the MethodDescriptor. <br> **Code Suggestion**: 
 ```
 Integrate strict request data validation before loading the MethodDescriptor to ensure only authorized access. Include checks for user authorization in relation to the requested MethodDescriptor.
 ```